### PR TITLE
Core/Misc Elite Tauren Chieftain Concert

### LINKED ISSUE #27850: DB/Misc Elite Tauren Chieftain Shattrath
### Description

The song "Power of The Horde" (id=181075) is now played during the event with a distance dependent sound.

### SQL fix

```sql
INSERT INTO `event_scripts` (`id`, `command`, `datalong`, `datalong2`, `x`, `y`, `z`, `o`) VALUES ('47', '16', '181075', '3', '-1750', '5136', '-36', '1.85');
```

### Branch

3.3.5

### TC rev. hash/commit

9d654345b50d

### PR DESCRIPTION
**Pull request by Plep-m**
**Changes proposed:**

- The music of the concert can be heard even if you were not there at the start of the concert
-  
-  

**Issues addressed:**
DB/Misc Elite Tauren Chieftain Shattrath #27850
Closes #27850 

**Tests performed:**

it build and i tested in game

**Known issues and TODO list:** (add/remove lines as needed)

- [ √]  hear the music even after the start of the concert
- [ ]  fix position of npc that currently still not on the stage at every concert place (sometime they are under the stage)

### CODE REVIEW

**Comment by Jinnaix** [comment]
Feels like something is missing here.

You are writing a gameobject script (go_etc_concert_music) but you do not attach this script to a gameobject via sql.
You need to set this in "ScriptName" for GameObject 186312 in gameobject_template table.

Follow up to-do - this object is not spawned. I dont know if we have a sniff for the correct placement. But if the object is not spawned in the world, nothing will happen.

Follow up to-do - this object needs to be handled in game_event_gameobject table for event 81, otherwise it would be constantly active. Additionally you can also set a break condition in the event and check for !IsHolidayActive(eventID) to prevent execution while the concert is not running.

**Comment by Plep-m** [comment]
okay thx im on it

File: src/server/scripts/World/go_scripts.cpp [hunk 1]
```diff
@@ -1212,7 +1212,55 @@ class go_bells : public GameObjectScript
...
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_ETC_START_MUSIC:
 
+                    me->PlayDirectMusic(POWER_OF_THE_HORDE);
```
Comment (Jinnaix) [inline src/server/scripts/World/go_scripts.cpp:1247]:
```suggestion
                case EVENT_ETC_START_MUSIC:
                    me->PlayDirectMusic(POWER_OF_THE_HORDE);
```
unnecessary line break here

File: src/server/scripts/World/go_scripts.cpp [hunk 2]
```diff
@@ -1212,7 +1212,55 @@ class go_bells : public GameObjectScript
         return new go_bellsAI(go);
     }
 };
+/*####
+## go_ETC70_music
+####*/
+
+enum ETC70Music {
+    POWER_OF_THE_HORDE = 181075,
+};
```
Comment (Jinnaix) [inline src/server/scripts/World/go_scripts.cpp:1221]:
```suggestion
enum ETC70Music 
{
    POWER_OF_THE_HORDE = 181075
};
```
Cosmetic change

File: src/server/scripts/World/go_scripts.cpp [hunk 3]
```diff
@@ -1212,7 +1212,56 @@ class go_bells : public GameObjectScript
...
     }
 };
+/*####
+## go_ETC70_music
+####*/
+
+enum ETC70Music 
+{
+    POWER_OF_THE_HORDE = 181075
+};
```
Comment (Jinnaix) [inline src/server/scripts/World/go_scripts.cpp:1222]:
```suggestion
enum ETC70Music
{
    POWER_OF_THE_HORDE = 181075
};
```
My fault, stupid code suggestion added a whitespace here that needs to be removed

**Comment by Riztazz** [comment]
This could be modernized a bit;
https://gist.github.com/Riztazz/b22d08054cd1533620b4107f3c87ab43

File: src/server/scripts/World/go_scripts.cpp [hunk 4]
```diff
@@ -1212,7 +1212,56 @@ class go_bells : public GameObjectScript
         return new go_bellsAI(go);
     }
 };
+/*####
```
Comment (Aokromes) [inline src/server/scripts/World/go_scripts.cpp:1215]: missing newline before /*###

File: src/server/scripts/World/go_scripts.cpp [hunk 5]
```diff
@@ -1212,7 +1212,56 @@ class go_bells : public GameObjectScript
...
+
+enum ETC70Music 
+{
+    POWER_OF_THE_HORDE = 181075
+};
+enum ETC70MusicEvent
+{
+    EVENT_ETC_START_MUSIC = 1
+};
+class go_etc_concert_music : public GameObjectScript
```
Comment (Aokromes) [inline src/server/scripts/World/go_scripts.cpp:1227]: missing newlibe before class

File: src/server/scripts/World/go_scripts.cpp [hunk 6]
```diff
@@ -1212,7 +1212,56 @@ class go_bells : public GameObjectScript
...
 };
+/*####
+## go_ETC70_music
+####*/
+
+enum ETC70Music 
+{
+    POWER_OF_THE_HORDE = 181075
+};
+enum ETC70MusicEvent
```
Comment (Aokromes) [inline src/server/scripts/World/go_scripts.cpp:1223]: missing newline before enum.

File: src/server/scripts/World/go_scripts.cpp [hunk 7]
```diff
@@ -1212,7 +1212,56 @@ class go_bells : public GameObjectScript
...
+    private:
+        EventMap _events;
+    };
+
+    GameObjectAI* GetAI(GameObject* go) const override
+    {
+        return new go_etc_concert_musicAI(go);
+    }
+};
 void AddSC_go_scripts()
```
Comment (Aokromes) [inline src/server/scripts/World/go_scripts.cpp:1265]: missing newline before void.

**Comment by jackpoz** [comment]
>[offl](https://github.com/offl) added the [Closed-ToBeImplementedDifferently](https://github.com/TrinityCore/TrinityCore/labels/Closed-ToBeImplementedDifferently) label [1 hour ago](https://github.com/TrinityCore/TrinityCore/pull/27875#event-6291720512)

did you have in mind a different way to implement this ?

**Comment by Plep-m** [comment]
Well I first tried to implement it directly to the database but someone tell me to implement it like the midsummer event so I tried

**Comment by Jinnaix** [comment]
> i supose [#27875 (comment)](https://github.com/TrinityCore/TrinityCore/pull/27875#issuecomment-1075977446)

Thats just a different way of coding the same thing and im sure it was just a hint from @Riztazz to adapt it.

@offl - any comment to this? Closing PRs from first time contributors without leaving a understandable reason is a questionable move.

**Comment by jackpoz** [comment]
> [@offl](/offl) [offl](/offl) added the [ Closed-ToBeImplementedDifferently ](/TrinityCore/TrinityCore/labels/Closed-ToBeImplementedDifferently) label [on Mar 23](#event-6291720512)
> [@offl](/offl) [offl](/offl) closed this [on Mar 23](#event-6291721864)

reopening this PR until we get some more details about why it was closed

**Comment by offl** [comment]
This is far more complex. You can take care of this PR, if you want. Or guide OP to change it. E.g. replace object entry with actual sound entry, create action in SAI for music(optionally), add scriptname to object to actually use it, remove Play Sound from SAI script, then spawn that object at every event position, link it to event, make timer match current timer in SAI. Ideally sniff at least one full event.

<details>

Some wip stuff

```cpp
    SMART_ACTION_UNIT_PLAY_MUSIC                    = 149,    // NYI
    SMART_ACTION_UNIT_PLAY_SOUND                    = 150,    // NYI
    SMART_ACTION_UNIT_PLAY_MUSIC_ZONE               = 151,    // NYI
    SMART_ACTION_UNIT_PLAY_SOUND_ZONE               = 152,    // NYI
    SMART_ACTION_UNIT_PLAY_TARGETED_SOUND           = 153,    // NYI Edge of TBC
    SMART_ACTION_UNIT_PLAY_TARGETED_MUSIC           = 154,    // NYI Edge of TBC, spells 45778, 45784
    SMART_ACTION_OBJECT_PLAY_MUSIC                  = 155,    // MusicID, basic implementation, too many things are unknown in sound\music actions
    SMART_ACTION_OBJECT_PLAY_SOUND                  = 156,    // NYI
    SMART_ACTION_OBJECT_PLAY_MUSIC_ZONE             = 157,    // NYI
    SMART_ACTION_OBJECT_PLAY_SOUND_ZONE             = 158,    // NYI

        case SMART_ACTION_OBJECT_PLAY_MUSIC:
        {
            for (WorldObject* target : targets)
                if (IsGameObject(target))
                    target->ToGameObject()->PlayDirectMusic(e.action.musicOrSound.entry);
            break;
        }

        struct
        {
            uint32 entry;
        } musicOrSound;
```

```sql
--
UPDATE `gameobject_template` SET `AIName` = 'SmartGameObjectAI' WHERE `entry` = 186312;
DELETE FROM `smart_scripts` WHERE `entryorguid` = 186312 AND `source_type` = 1;
INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(186312,1,0,0,60,0,100,0,9400,9400,5000,5000,0,155,11803,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"L70ETC Music Doodad - On Update - Play Music 11803");

DELETE FROM `gameobject` WHERE `guid` = X AND `id` = 186312;
INSERT INTO `gameobject` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `position_x`, `position_y`, `position_z`, `orientation`, `rotation0`, `rotation1`, `rotation2`, `rotation3`, `spawntimesecs`, `animprogress`, `state`, `ScriptName`, `VerifiedBuild`) VALUES
(X,186312,530,0,0,1,1,-1751.67,5140.7,-36.2613,4.99164,0,0,-0.601814,0.798636,300,255,1,'',15595);
```

**Pull request closed without merging**